tools/ds-identify

```diff
@@ -1409,33 +1376,31 @@ dscheck_VMware() {
 
     # Check to see if running in a container and the VMware
     # datasource is configured via environment variables.
-    if vmware_has_envvar_vmx_guestinfo; then
-        if vmware_has_envvar_vmx_guestinfo_metadata || \
-            vmware_has_envvar_vmx_guestinfo_userdata || \
-            vmware_has_envvar_vmx_guestinfo_vendordata; then
+    if [ -n "${VMX_GUESTINFO}" ]; then
```

Comment:
@akutz - do we need to use a new function for each of these checks? I thought otherwise and kept it simple.
If you think it's needed, please let me know. It will be good to know the reason as well.

Reply:
The new functions make it possible to test in [`test_ds-identify.py`](https://github.com/canonical/cloud-init/blob/main/tests/unittests/test_ds_identify.py). Please change everything back to the way it was. Thanks!

Reply:
Also, these were net-new and I'm not sure what the thinking was removing them. In the future reaching out on Slack or async is probably a better path forward before making so many reactive changes. At least have a conversation first. I'm nothing if not talkative :)

Reply:
Also, FWIW, the way you changed it was the way I had it in the original repo. As I indicated, the new model was to support the testing framework for ds-identify.